### ISSUE TITLE
box_usage_linter: permit use of double colon operator?

### ISSUE THREAD

**Issue by tyner** [Appsilon/box.linters#123]
### box.linters version

0.9.1

### Sample source code to lint

```r
graphics::plot(1, 1)
```

### Lint command used

```r
lintr::lint(
  "/path/to/test_file.R",
  linters = box.linters::box_default_linters["box_usage_linter"]
)
```

### Lint result

```
/path/to/test_file.R:1:1: warning: [box_usage_linter] Function not imported nor defined.
graphics::plot(1, 1)
^~~~~~~~~~~~~~
```

### Expected result

It would be nice if this did not trigger a lintr warning, as the double colon operator is quite standard and fully compatible with the `box` paradigm. Thank you for considering.

**Comment by caldwellst** [comment]
Just jumping in as another potential user of `{box.linters)` and frequent `{box}` user. I think the current behaviour is exactly what is expected, and the use of `graphics::plot()` does not fit the `{box}` paradigm. `{box}` requires explicit importing of all modules outside of what is in `{base}`. So, if I were using `{box}`, I would expect that to be:

```r
box::use(graphics)

graphics$plot(1, 1)
```

From the [migration vignette](https://cran.r-project.org/web/packages/box/vignettes/migration.html):

> ‘box’ modules no longer attach any packages by default, except ‘base’. If a module needs to use functions from those other packages, it needs to declare them explicitly (e.g. via `box::use(stats)`). If users want to import all default R packages, they can import the module r/core for this purpose. The following declaration approximates the behaviour of ‘modules’:
> 
> ```r
> box::use(r/core[...])
> ```

So you could also just do that and use `plot()` directly as you would in a standard R session. But as is, you have not explicitly imported `{graphics}`, and just directly used it somewhere in your script, so is breaking the key use of `{box}` to ensure explicit importation a la Python and other programming languages.

All of this to say that I would vote to not implement the change above, as flagging attachments using `module::function()` is a key way of ensuring that `box::use()` has been utilised correctly.

**Comment by tyner** [comment]
@klmr do you have an opinion? To me, using the double colon operator fits squarely within the 'explicit is better than implicit' philosophy to which `box` aspires.

**Comment by klmr** [comment]
This is definitely a judgement call, but my personal design philosophy with ‘box’ is that it’s a holistic solution that *replaces* other ways of interacting with packages — for consistency if for no other reason.

In fact, `source()`, `library()` and `require()` calls inside a ‘box’ module cause a warning message by default; it’s conceivable that using `::` will also cause a warning in the future do this (the fact that it currently doesn’t is actually an oversight).

**Comment by tyner** [comment]
Tested with version 0.10.2 and it looks good. Thank you!
